contents/scalability/scalable-development.md

```diff
@@ -33,29 +33,29 @@ There are a few common bottlenecks that can be avoided such as database querying
+The next thing to consider is what to cache. The rule of thumb is to cache data that is frequently accessed and read more often than it is updated. With this, cache hits would be more often than cache misses and as a result, the speed increase would outweigh the cost of maintaining the cache.  
 
-Caching ([resource](http://www.lecloud.net/post/9246290032/scalability-for-dummies-part-3-cache))
+1. A beginner's guide to scalability using caches which also describes in more detail on what to cache and how to do so ([resource](http://www.lecloud.net/post/9246290032/scalability-for-dummies-part-3-cache))
 
-Scalability best practices ([resource](https://www.infoq.com/articles/ebay-scalability-best-practices))
+1. Scalability best practices ([resource](https://www.infoq.com/articles/ebay-scalability-best-practices))
 
 ### Go Asynchronous 
-Unlike synchronous operations, which runs sequentially and waits for the previous operation to complete before moving on to the next, asynchronous operations don't wait for other operations to finish before proceeding. Instead, they continue and allow other operations to finish.
+Unlike synchronous operations, which runs sequentially and waits for the previous operation to complete before moving on to the next, asynchronous operations run in parallel, so they do not wait for prior operations to finish before running. Instead, they continue and allow other operations to finish.
 
-By using asynchronous operations, components are less tightly coupled and can be scaled independently of one another. As a result, if one component is down, the other component can still move forward.
+If one component A calls another component B synchronously, they are tightly coupled as both components depend on each other. As a result, to scale A, you must also scale B. By using asynchronous operations, components are decoupled and can be scaled independently of one another.
 
-In situations in which responding to request is crucial, asynchronous operations can reduce the latency experienced by the requester. This is done by prioritising the quickness of the response to the user over the speed in which other processes complete.
+In situations in which responding to request is crucial, asynchronous operations can reduce the latency experienced by the requester. This is done by prioritising the quickness of the response to the user over the speed of other processes such as execution latency (how quickly we get everything done). For instance, rather than waiting for some processes such as tracking the user's activity for recommending products in the near future to finish, it's better to asynchronously update the user interface and then finish these processes. In this way, the user experiences lesser latency.
```

Comment:
> such as execution latency (how quickly we get everything done)

Is `(how quickly the request is done executing)` better?

> For instance, rather than waiting for some processes such as tracking the user's activity for recommending products in the near future to finish

Maybe we can use a simpler example, such as `downloading file`? So while the file downloads, your ui doesn't hang.

Reply:
>Is (how quickly the request is done executing) better?

`request is done executing` sounds a bit off though.  Maybe `how quickly the request is processed`?